pandas/core/nanops.py

```diff
@@ -1003,11 +1003,23 @@ def nanvar(
     # cancellation errors and relatively accurate for small numbers of
     # observations.
     #
-    # See https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
-    avg = _ensure_numeric(values.sum(axis=axis, dtype=np.float64)) / count
+    # See https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance...
+    if values.dtype.kind == "c":
+        avg = _ensure_numeric(values.sum(axis=axis, dtype=values.dtype)) / count
+    else:
+        avg = _ensure_numeric(values.sum(axis=axis, dtype=np.float64)) / count
     if axis is not None:
         avg = np.expand_dims(avg, axis)
-    sqr = _ensure_numeric((avg - values) ** 2)
+    # ...but also, see https://numpy.org/doc/stable/reference/generated/numpy.nanvar.html#numpy-nanvar,  # noqa: E501
```

Comment:
Could you format these to avoid using `# noqa: E501` (and likewise for the rest)